toptal/picasso
# [FX-5726] Add base tailwind preset package

[FX-5726]

### Description

This pull request introduces the `@toptal/base-tailwind` package that provide base Tailwind configuration for any project. Configurations that are specific to projects are kept in the corresponding packages (for example, `@toptal/picasso-tailwind`).

### How to test


- [Temploy](https://picasso.toptal.net/fx-5726-create-base-preset-package)
- The https://github.com/toptal/staff-portal/pull/13505 applies both presets (please launch Staff Portal locally if needed). As you can see, the pull request temporary moves `fontWeight` to `@toptal/base-tailwind`, so we can see new font weight `700` to be applied from `@toptal/base-tailwind`.

<img width="1287" alt="Screenshot 2024-08-09 at 17 47 06" src="https://github.com/user-attachments/assets/908c74d8-90c1-4cf8-a46a-e5c59ec8b9c9">

Some other styles are broken, as `fontWeight` is missing in `@toptal/picasso-tailwind` that some components rely on. 

### Development checks

- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-5762]: https://toptal-core.atlassian.net/browse/FX-5762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-5726]: https://toptal-core.atlassian.net/browse/FX-5726?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[B2-732]: https://toptal-core.atlassian.net/browse/B2-732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
